cpp/change-notes/2021-09-13-overflow-static.md

```diff
@@ -0,0 +1,6 @@
+lgtm,codescanning
+* The `Buffer` library considers more fields to be of variable size
+  for array members of size 0 or 1. Buffer size calculation of array type
+  fields of size 0 or 1 in unions are considered pointers to the union
+  and will return the size of the union itself. The changes reduces
+  the number of false positives in cpp/static-buffer-overflow
```

Comment:
I think I'd phrase it in _slightly_ less detail:
```
* The `memberMayBeVarSize` predicate considers more fields to be variable size.
  As a result, the "Static buffer overflow" query (cpp/static-buffer-overflow)
  produces fewer false positives.
```
What do you think of this one?

Reply:
Sounds good. I did also change buffer size calculcation on arrays of size 0 or 1 in unions - is that worth mentioning?

Reply:
I think it's fine to leave that detail out. That's also a change that reduces the false positive rate of `cpp/static-buffer-overflow`, right? If so, I think we can sweep that detail under the rug.

Reply:
I will update the changenote